### PR TITLE
feat: add docker testing

### PR DESCRIPTION
**Pull request by spion**
This PR adds a scaffold for automated tests based on docker and an android emulator. The neat thing about this approach is that we can also use multiple versions of android, and while that may not be fully exhaustive in terms of the devices out there, its way better than having nothing.

Without tests, changes to this project are quite high risk. With tests, I think we can get things going :smile: 

Currently covers only a single Android version, but its easy to add a second dockerfile.

To try it out

```
docker build -t adbfs-rootless-v1 -f docker/Dockerfile .
docker run --privileged --rm adbfs-rootless-v1
```

The reason `--privileged` is needed is because the android emulators need virtualization support (/dev/kvm needs to be available). I'm still hoping we'll be able to run these in GitHub actions but I've not tried yet.

This is WIP. Would be great to have some ideas about good representative tests to run and ways to verify that everything is ok.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# FROM ubuntu:22.04 as buildenv
```
Comment (spion) [inline docker/Dockerfile:1]: I can revive this and build in a separate container, we just need 18.04 instead as that's what budtmo/docker-android is based on.

**Comment by spion** [comment]
For running priviledged actions, refer to https://github.com/actions/container-action/issues/2#issuecomment-854057781

**Comment by spion** [comment]
Looks like this is not happening with github actions for now: https://github.com/actions/runner-images/issues/183

**Comment by spion** [comment]
TODO: check if this helps https://www.ubicloud.com/use-cases/github-actions

**Comment by spion** [comment]
Its so I can also run the tests locally (with multiple android versions, as well as multiple versions of ubuntu/fuse/etc) without having to rely on GHA. But I'm probably going to be reworking that bit to use separate containers for the emulator and the filesystem "host"

**Comment by ViliusSutkus89** [comment]
The way I see it, you need docker to be able to test locally on different ubuntu/fuse versions. Just because you need that, it doesn't mean that the script should ONLY be callable through docker. You could use docker locally, and plain naked script in GHA. Both still calling the same adb calls. With GHA you get an easy way to set up a test matrix with all relevant android versions and maybe even fuse versions, if different fuse versions are easily reachable through apt-get or whatever.

Keep in mind that GHA Android emulators are x86/x86_64. I haven't tried anything with the new macos armv8 images, maybe they would allow armv8 emulator images in addition to ubuntu's x86/x86_64.

Let me know if you want help here on the emulator or GHA

**Comment by spion** [comment]
Yeah, the script is already called directly in GHA now from an action that sets up the emulator. The code is a year old and went through quite a few iterations - my main goal yesterday was to get it working in case there is a new influx of contributions but there is some cleanup left to do for sure :sweat_smile:

**Pull request merged**